src/UtilityTrait.php

```diff
@@ -446,4 +453,37 @@ private function intersectArrayRecursively($master, $mask)
         }
         return $master;
     }
+
+    /**
+     * Appends query param at the end of url.
+     *
+     * @param string $url
+     * @param string $varName
+     * @param string $value
+     */
+    function appendQueryParam(&$url, $varName, $value)
```

Comment:
don't use references (`immutability over mutability`) and instead return new url